Binary files wxorig/src/osx/.DS_Store and wxnew/src/osx/.DS_Store differ
diff -uNr wxorig/src/osx/core/printmac.cpp wxnew/src/osx/core/printmac.cpp
--- wxorig/src/osx/core/printmac.cpp	2009-10-08 10:06:04.000000000 -0400
+++ wxnew/src/osx/core/printmac.cpp	2009-11-27 12:31:04.000000000 -0500
@@ -449,18 +449,6 @@
 
     printout->SetIsPreview(false);
 
-    // Get some parameters from the printout, if defined
-    int fromPage, toPage;
-    int minPage, maxPage;
-    printout->GetPageInfo(&minPage, &maxPage, &fromPage, &toPage);
-
-    if (maxPage == 0) return false;
-
-    // Only set min and max, because from and to will be
-    // set by the user
-    m_printDialogData.SetMinPage(minPage);
-    m_printDialogData.SetMaxPage(maxPage);
-
     if (m_printDialogData.GetMinPage() < 1)
         m_printDialogData.SetMinPage(1);
     if (m_printDialogData.GetMaxPage() < 1)
@@ -515,6 +503,22 @@
 
     printout->OnPreparePrinting();
 
+    // Get some parameters from the printout, if defined
+    int fromPage, toPage;
+    int minPage, maxPage;
+    printout->GetPageInfo(&minPage, &maxPage, &fromPage, &toPage);
+
+    if (maxPage == 0)
+    {
+        sm_lastError = wxPRINTER_ERROR;
+        return false;
+    }
+
+    // Only set min and max, because from and to have been
+    // set by the user
+    m_printDialogData.SetMinPage(minPage);
+    m_printDialogData.SetMaxPage(maxPage);
+
     printout->OnBeginPrinting();
 
     bool keepGoing = true;
